code/datums/components/_component.dm

```diff
@@ -310,18 +308,11 @@
 /datum/proc/_SendSignal(sigtype, list/arguments)
 	var/target = comp_lookup[sigtype]
 	if(!length(target))
-		var/datum/C = target
-		if(!(C.datum_flags & DF_SIGNAL_ENABLED))
-			return NONE
-		var/proctype = C.signal_procs[src][sigtype]
-		return NONE | CallAsync(C, proctype, arguments)
+		var/datum/listening_datum = target
+		return NONE | CallAsync(listening_datum, listening_datum.signal_procs[src][sigtype], arguments)
```

Comment:
Why do `NONE | ` at all? Just to annotate its a bitfield? We can do that with a comment (but also, anyone looking at this code should know that already). This isn't slow by any means, but it's wasteful.

Reply:
cant find a real reason so its gone

Reply:
`NONE` is `0` and without this it can return null, it likely doesn't affect much but nonetheless this is a behavior change and I'd prefer to have signals always return a number as it's supposed to represent flags.